### PR TITLE
AEA-500 Add agent context namespace.

### LINKED ISSUE #1008: Provide option to add objects to AEA constructor
**Is your feature request related to a problem? Please describe.**
AEA constructor does not allow for kwargs

**Describe the solution you'd like**
Allow for kwargs which are then attached to the agent context and exposed on each skill context.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

The agent context namespace is populated via either AgentContext constructor or AEA constructor through keyword arguments.

It can be accessed with
- agent_context.namespace.attribute_name
- skill_context.namespace.attribute_name

## Fixes

Fixes #1008 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [X] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [X] Lint and unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [X] I have added necessary documentation (if appropriate)
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/aea_builder.py [hunk 1]
```diff
@@ -244,6 +244,7 @@ def __init__(self, with_default_packages: bool = True):
             "fetchai"  # set by the user, or instantiate a default one.
         )
         self._default_connection = PublicId("fetchai", "stub", "0.1.0")
+        self.context_namespace = {}  # type: Dict[str, Any]
```
Comment (DavidMinarsch) [inline aea/aea_builder.py:247]: Shouldn't the builder provider a setter method?

File: tests/test_aea.py [hunk 2]
```diff
@@ -436,8 +436,6 @@ def setup_class(cls):
         resources = Resources()
         resources.add_component(Skill.from_dir(Path(CUR_PATH, "data", "dummy_skill")))
         identity = Identity(agent_name, address=wallet.addresses[FETCHAI])
-        cls.input_file = tempfile.mkstemp()[1]
```
Comment (DavidMinarsch) [inline tests/test_aea.py:439]: Why are they removed?

Comment (marcofavorito) [reply tests/test_aea.py:439]: because they were not needed (we use OEFLocalConnection)

File: docs/skill.md [hunk 3]
```diff
@@ -16,6 +16,8 @@ This means it is possible to, at any point, grab the `context` and have access t
 
 For example, in the `ErrorHandler(Handler)` class, the code often grabs a reference to its context and by doing so can access initialised and running framework objects such as an `OutBox` for putting messages into.
 
+Moreover, you can read/write to the _agent context namespace_ by accessing the attribute `SkillContext.namespace`.
+
```
Comment (marcofavorito) [inline docs/skill.md:20]: Too short? :-)

Comment (DavidMinarsch) [reply docs/skill.md:20]: Ok, maybe should also explain how one feeds objects to it via the builder?

**Pull request merged**